src/ilcdlib/entity/exchage.py

```diff
@@ -0,0 +1,82 @@
+#
+#  Copyright 2023 by C Change Labs Inc. www.c-change-labs.com
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  This software was developed with support from the Skanska USA,
+#  Charles Pankow Foundation, Microsoft Sustainability Fund, Interface, MKA Foundation, and others.
+#  Find out more at www.BuildingTransparency.org
+#
+from openepd.model.lcia import OutputFlowSet, ResourceUseSet, ScopeSet
+
+from ilcdlib.entity.base_scope_set_reader import BaseIlcdScopeSetsReader
+from ilcdlib.mapping.common import SimpleDataMapper
+from ilcdlib.mapping.flows import default_flows_uuid_mapper
+from ilcdlib.mapping.indicators import default_indicators_uuid_mapper
+from ilcdlib.xml_parser import T_ET
+
+
+class IlcdExchangesReader(BaseIlcdScopeSetsReader):
+    """Read exchanges from XML file."""
+
+    def __init__(
+        self,
+        *args,
+        indicator_mapper: SimpleDataMapper[str] = default_indicators_uuid_mapper,
+        flow_mapper: SimpleDataMapper[str] = default_flows_uuid_mapper,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.indicator_mapper = indicator_mapper
+        self.flow_mapper = flow_mapper
+
+    def __get_exchange_direction(self, el: T_ET.Element) -> str | None:
+        exchange_direction = self._get_el(el, "process:exchangeDirection")
+        if exchange_direction is not None:
+            return exchange_direction.text
+        return None
+
+    def get_exchanges(self) -> tuple[ResourceUseSet, OutputFlowSet]:
```

Comment:
I don't like the fact we make this reader too much bound to the openEPD use. The overall idea of this library design is to make a useful API for ILCD format. openEPD conversion is just a nice capability. However this particular reader is designed to return something very specific to openEPD. 

Let's have a call today and see if we can come up with something better.

Reply:
Updated